src/activities/drawnumber/drawnumber.js

```diff
@@ -137,50 +240,93 @@ function drawSegment(pointIndex) {
+        items.pointImageRepeater.itemAt(0).highlight = true
+        // prepare segments data
+        linePropertiesArray = []
+        for (var i = 0; i < (pointPositions.length)-1; i++) {
+            var lineProperties = []                   // properties are x1,y1,x2,y,angle rotation
+            lineProperties[0] = pointPositions[i][0]                                    // x
+            lineProperties[1] = pointPositions[i][1]                                    // y
+            lineProperties[2] = pointPositions[i+1][0]                                  // x2
+            lineProperties[3] = pointPositions[i+1][1]                                  // y2
+            linePropertiesArray[i] = lineProperties
+        }
+        items.segmentsRepeater.model = linePropertiesArray
     }
 
-    // prepare segments data
-    linePropertiesArray = []
-    for (var i = 0; i < (pointPositions.length)-1; i++) {
-        var lineProperties = []                   // properties are x1,y1,x2,y,angle rotation
-        lineProperties[0] = pointPositions[i][0]                                    // x
-        lineProperties[1] = pointPositions[i][1]                                    // y
-        lineProperties[2] = pointPositions[i+1][0]                                  // x2
-        lineProperties[3] = pointPositions[i+1][1]                                  // y2
-        linePropertiesArray[i] = lineProperties
-    }
-    items.segmentsRepeater.model = linePropertiesArray
+    else if(mode=="clickanddraw" || mode=="drawnumber"){
 
+        // prepare points data
+        pointPositions = dataset[currentLevel].coordinates
+        items.pointImageRepeater.model = pointPositions
+
+        if(mode == "clickanddraw"){
+        items.pointImageRepeater.itemAt(0).highlight = true
+        }
+
+        // prepare segments data
+        linePropertiesArray = []
+        for (var i = 0; i < (pointPositions.length)-1; i++) {
+            var lineProperties = []                   // properties are x1,y1,x2,y,angle rotation
+            lineProperties[0] = pointPositions[i][0]                                    // x
+            lineProperties[1] = pointPositions[i][1]                                    // y
+            lineProperties[2] = pointPositions[i+1][0]                                  // x2
+            lineProperties[3] = pointPositions[i+1][1]                                  // y2
+            linePropertiesArray[i] = lineProperties
+        }
+        items.segmentsRepeater.model = linePropertiesArray
+    }
 }
 
 function loadBackgroundImage() {
-    items.imageBack.source = url + dataset[currentLevel].imageName1
+
+    if(mode == "drawletters"){
+        items.imageBack.source = url + dataset2[currentLevel].imageName1
+    }
+
+    else if(mode == "clickanddraw" || mode == "drawnumber"){
+        items.imageBack.source = url + dataset[currentLevel].imageName1
+
+    }
 }
 
 function won() {
+    if(mode == "drawletters"){
```

Comment:
if you have the sound directly in the dataset, you should just check if the property 'sound' exists and play it.